### PR TITLE
Ae refactor

### PR DESCRIPTION
**Pull request by paetling**
review @thieman

### CODE REVIEW

File: dusty/compiler/compose/__init__.py [hunk 1]
```diff
@@ -5,11 +5,64 @@
...
+def lib_install_commands_for_app_or_lib(app_or_lib_name, assembled_specs):
+    if app_or_lib_name in assembled_specs['apps']:
+        return _lib_install_commands_for_app(app_or_lib_name, assembled_specs)
+    return _lib_install_commands_for_lib(app_or_lib_name, assembled_specs)
+
+def _lib_install_command(lib_spec):
+    """ This returns a single commmand that will install a library in a docker container """
+    if not lib_spec['install']:
+        return ''
+    return "cd {}; {}".format(lib_spec['mount'], '; '.join(lib_spec['install']))
```
Comment (paetling) [inline dusty/compiler/compose/__init__.py]: temporarily left in here

File: dusty/compiler/compose/__init__.py [hunk 2]
```diff
@@ -46,14 +99,14 @@ def _conditional_links(assembled_specs, app_name):
...
-    """ Given a spec for an app, returns the value of the `build` field for docker-compose.
+    """ Given a spec for an app, returns the value of the `build` field for dockercompose.
     If the path is relative, it is expanded and added to the path of the app's repo. """
     if os.path.isabs(app_spec['build']):
         return app_spec['build']
     return os.path.join(Repo(app_spec['repo']).local_path, app_spec['build'])
 
 def _composed_app_dict(app_name, assembled_specs, port_specs):
-    """ This function returns a dictionary of the docker-compose.yml specifications for one app """
+    """ This function returns a dictionary of the dockercompose.yml specifications for one app """
```
Comment (paetling) [inline dusty/compiler/compose/__init__.py]: going to fix these.

File: dusty/compiler/compose/__init__.py [hunk 3]
```diff
@@ -5,11 +5,64 @@
 
 from ..spec_assembler import get_assembled_specs
 from ...path import vm_cp_path
-from ...source import Repo
 from ... import constants
+from ...command_file import dusty_command_file_name
+from .helper_functions import container_code_path, get_volume_mounts, get_app_volume_mounts, get_lib_volume_mounts
```
Comment (thieman) [inline dusty/compiler/compose/__init__.py]: Should this be `common`?

File: dusty/systems/docker/testing_image.py [hunk 4]
```diff
@@ -2,8 +2,10 @@
 
 import docker
 
-from ...compiler.compose import lib_install_commands_for_app_or_lib, container_code_path, get_volume_mounts
+from ...compiler.compose import container_code_path, get_volume_mounts
 from ...log import log_to_client
+from ...command_file import dusty_command_file_name, lib_install_commands_for_app_or_lib
+from .common import spec_for_service
```
Comment (thieman) [inline dusty/systems/docker/testing_image.py]: This diff doesn't make sense if nothing else in the file changed

**Comment by thieman** [comment]
Seems legit

File: dusty/compiler/compose/__init__.py (hunk 3 above)
Comment (paetling) [inline dusty/compiler/compose/__init__.py]: yes.  how did this happen. hmm.

**Pull request merged**